core-jdk8/src/main/java/org/mapstruct/Mapping.java

```diff
@@ -58,7 +58,7 @@
      *
      * @return The target name of the configured property or enum constant
      */
-    String target();
+    String target() default "";
```

Comment:
I don't think we should make `target()` optional. The idea of `@Mapping` has always been that it describes a target property - so its name is mandatory - and then the different ways for how it could get its value (simple source property, expression, constant etc.).

Thinking about it with @sjaakd and @filiphr, a better approach would be to add `@BeanMapping#ignoredSourceProperties()`. That way the semantics of `@Mapping` aren't changed and you still get the feature you're after.

WDYT?

Reply:
Grumble, grumble, could’ve saved me a throw-away implementation by having that discussion earlier. Ok, now that’s out of the way... :)

Is there anything to say for a consistent approach to “ignore” mappings, supporting both source & target via either @BeanMapping or via @Mapping?

Also, with @BeanMapping is there any ambiguity between an ignored source property and a forged mapping? With @Mapping, ignored source should clearly prevent forgery.